backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/banking/api/BankRepository.java

```diff
@@ -160,4 +162,15 @@ public DataImportConfigId retrieveDataImportConfigIdForBank(@Nullable final Bank
 				this::retrieveDefaultBankDataImportConfigId);
 	}
 
+	@NonNull
+	public Set<BankId> retrieveBankIdsByName(final String bankName)
+	{
+		return queryBL.createQueryBuilder(I_C_Bank.class)
+				.addStringLikeFilter(I_C_Bank.COLUMNNAME_Name, bankName, false)
+				.addOnlyActiveRecordsFilter()
+				.iterateAndStream()
+				.map(I_C_Bank::getC_Bank_ID)
+				.map(BankId::ofRepoId)
+				.collect(ImmutableSet.toImmutableSet());
+	}
```

Comment:
- doesn't make much sense to use iterateAndStream if we are anyway collecting all results in a Set,  you could use directly `stream()` in such scenarios
- in this particular case you can directly use `.listIds()`